crates/wallet/src/errors/erros.rs

```diff
@@ -0,0 +1,40 @@
+use thiserror::Error;
+use tonic::Status;
+use nuts::nut00::NutError;
+use nuts::nut01::KeyError;
+use nuts::nut02::KeysetIdError;
+
+#[derive(Error, Debug)]
+pub enum WalletError {
+    #[error("Database error: {0}")]
+    Database(#[from] rusqlite::Error),
+
+    #[error("Transport error: {0}")]
+    Transport(#[from] tonic::transport::Error),
+    
+    #[error("gRPC error: {0}")]
+    Grpc(#[from] Status),
+
+    #[error("Nut error: {0}")]
+    Nut(#[from] NutError),
+
+    #[error("Key error: {0}")]
+    Key(#[from] KeyError),
+
+    #[error("Keyset ID error: {0}")]
+    KeysetId(#[from] KeysetIdError),
+
+    #[error("Amount overflow")]
+    AmountOverflow,
+
+    #[error("No matching keyset found")]
+    NoMatchingKeyset,
+
+    #[error("Proof not available")]
+    ProofNotAvailable,
+
+    #[error("Invalid public key: {0}")]
+    InvalidPublicKey(String),
```

Comment:
no need for all this spacing between the variants

Reply:
i thought it's more easier to read than this 

<img width="538" alt="Screenshot 2025-03-11 at 09 23 43" src="https://github.com/user-attachments/assets/20df6f41-6302-476d-8d2e-0b392fec05d5" />